Implement Profile Listings Page

rj13371 opened the pull request:
### What this PR does :
Made the skeleton for profile listings page, I have made fake profile objects and am rendering them until I can complete the profile listings integration ticket. 

### Screenshots / Videos :
![demo screenshot](https://i.imgur.com/RUvp4ot.png)
![demo screenshot mobile](https://i.imgur.com/MnlowM0.png)

### Any information needed to test this feature :
You can view the route on http://localhost:3000/profile-listings 
But normally we would go to a dynamic route from the landing page that submits a search query of location, date in, date out , then redirects you to this page with the passed in props. 

The search bar is yet to be implemented.

### Any issues with the current functionality (optional):
Still having trouble with MUI's way of writing css so right now the Cards are not responsive on wide viewports and look awkward with lots of whitespace on the right. 

useStyles does not behave the way I thought it was originally intended for IE MUI overrides most classes you place unless you specifically target them or add them to the app theme which I obviously do not want to do for one component. I found out that the sx prop actually overrides MUI when you want it to so I am using it on a few components.

rj13371 committed:
Made skeleton for profile listings and fake profiles just to render profile cards until integration

rj13371 committed:
changing to useeffect and testing why it does not render any cards

rj13371 committed:
styling changes

rj13371 committed:
styling changes

rj13371 committed:
Made it responsive on Mobile but having difficulty on large monitors

rj13371 committed:
font styling changes

sundayezeilo commented on client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:
```diff
@@ -0,0 +1,137 @@
+  dropOff: Date;
+}
+
+//we have an interface Profile in another branch so this is just temporary, change this to import a Profile interface when its on main branch.
```
Remove comment, please.

sundayezeilo commented on client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:
```diff
@@ -0,0 +1,137 @@
+type Profiles = Profile[];
+
+export default function ProfileListing({}: Props): ReactElement {
+  //currently fake profiles for state
```
Comment not needed please

sundayezeilo commented on client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:
```diff
@@ -0,0 +1,137 @@
+  useEffect(() => {
+    // use this later when we can integrate, this is an example of getting an API resp and then setting state with it
+    //   getProfiles().then((res) => {
+    //   });
```
You can make a note of this in  your todos but not suitable here as it makes our code untidy :smiley:

sundayezeilo reviewed (commented):
Looks good! Please take a look at the few comments I left you and don't forget to link your ticket in your next PRs.
Happy coding!

rj13371 committed:
removed comments

rajivtitus commented on client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:
```diff
@@ -0,0 +1,130 @@
+  dropOff: Date;
+}
+
+interface Profile {
```
Why don't we move this interface to the 'interface' directory

rajivtitus commented on client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:
```diff
@@ -0,0 +1,130 @@
+      _id: '61df3960442e349d9200000',
+      photo: 'fakeimage',
+    };
+    const fakeProfiles = [profile, profile, profile, profile, profile];
```
👍🏽

rajivtitus commented on client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:
```diff
@@ -0,0 +1,130 @@
+          {profiles.length > 0 &&
+            profiles.map((profile) => (
+              <Grid key={profile.userId} item md={'auto'}>
+                <Card className={classes.profileCard}>
```
Good use of the MUI Card API 💯

rajivtitus commented on client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:
```diff
@@ -0,0 +1,130 @@
+                        {`${profile.title}`}
+                      </Typography>
+
+                      <Rating name="half-rating" defaultValue={2.5} precision={0.5} />
```
```suggestion
                      <Rating name="half-rating" defaultValue={2.5} precision={0.5} readOnly />
```

rajivtitus reviewed (commented):
Very good work with this Roland!
Let's also complete the following:
- Add text fields/date pickers for the search bar 
- Refine the padding for the UI
- Maybe use the substring method to only display a limited number of chars on each card
Thanks 👍🏽

rj13371 committed:
add read only to element

Co-authored-by: Rajiv Titus <75184965+rajivtitus@users.noreply.github.com>

rj13371 committed:
add substring to card

rj13371 committed:
added search bar with location and date. Moved profile to interface folder

rj13371 committed:
refined styles on cards for bigger viewports

rj13371 commented:
I refined the UI to look better on wider monitors, added a substring of 80 chars max, and added a placeholder form for location and date, but as I understand this is being made in the Dashboard integration ticket right? So this will be changed later?

rajivtitus commented:
> I refined the UI to look better on wider monitors, added a substring of 80 chars max, and added a placeholder form for location and date, but as I understand this is being made in the Dashboard integration ticket right? So this will be changed later?

We just needed the search bar UI in this PR, it will be made functional in the `FS: Dashboard Ticket`

rajivtitus reviewed (approved):
LGTM 👍🏽

rj13371 committed:
Merge branch 'main' of https://github.com/hatchways/team-pine into feat-profile-listings-fe

rj13371 committed:
change profile to fit listings
